crepecake/src/main/java/com/google/cloud/tools/crepecake/blob/BlobStream.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.crepecake.blob;
+
+import com.google.common.io.ByteStreams;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+/** A read-only {@link OutputStream} for BLOBs. */
+public class BlobStream {
```

Comment:
Can you link to the PR where this is used? It just seems like too simple a wrapper in this form.

Reply:
Yeah, it is used in #43 #44 and #45 currently.

The idea is to have this be the common datatype for passing around BLOBs that can be instantiated from various ways:

- from an InputStream - reading from a file or HTTP response
- from a String - from a JSON string
- from another OutputStream - the built layer tar stream

Is this sufficient for having this wrapper?

Reply:
I'm vert nervous about the design here. I think you should probably store a byte array instead, and convert it to a ByteArrayOutputStream as needed.